shared/src/commonMain/kotlin/ui/views/PresentationView.kt

```diff
@@ -0,0 +1,291 @@
+        PresentationStateModel.State.COMPLETED -> {
+            // Delay for a short amount of time so the user has a chance to see the success/error indication
+            coroutineScope.launch {
+                delay(1.5.seconds)
+                onPresentmentComplete()
+            }
+        }
+    }
+
+    if (state != PresentationStateModel.State.WAITING_FOR_DOCUMENT_SELECTION) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Center
+        ) {
+            Spacer(modifier = Modifier.weight(0.15f))
+            Column(
+                modifier = Modifier.fillMaxWidth(),
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                val (appNameText, iconPainter, iconCaptionText) = when (state) {
+                    PresentationStateModel.State.IDLE,
+                    PresentationStateModel.State.CONNECTING -> {
+                        Triple(
+                            stringResource(Res.string.app_display_name),
+                            painterResource(Res.drawable.icon_presentation),
+                            stringResource(Res.string.presentation_connecting_to_verifier)
+                        )
+                    }
+
+                    PresentationStateModel.State.WAITING_FOR_SOURCE,
+                    PresentationStateModel.State.PROCESSING
+                        -> {
+                        Triple(
+                            stringResource(Res.string.app_display_name),
+                            painterResource(Res.drawable.icon_presentation),
+                            if (presentationStateModel.numRequestsServed.collectAsState().value == 0) {
+                                ""
+                            } else {
+                                stringResource(Res.string.presentation_waiting_for_request)
+                            }
+                        )
+                    }
+
+                    PresentationStateModel.State.COMPLETED -> {
+                        when (presentationStateModel.error) {
+                            null -> Triple(
+                                    //TODO add proper icons
+                                    "", painterResource(Res.drawable.icon_presentation_success),
+                                    stringResource(Res.string.presentation_success)
+                                )
+                            is PresentmentCanceled -> Triple(
+                                    stringResource(Res.string.app_display_name),
+                                    painterResource(Res.drawable.icon_presentation),
+                                    stringResource(Res.string.presentation_canceled)
+                                )
+                            is PresentmentTimeout -> Triple(
+                                    "", painterResource(Res.drawable.icon_presentation_error),
```

Comment:
Please refactor those lines manually, so that one can clearly see the three parameters to `Triple`